sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -592,6 +596,19 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
         return new GlobalThroughputControlConfigBuilder(this, databaseId, containerId);
     }
 
+    List<OpenConnectionResponse> openConnectionsAndInitCaches() {
+        return openConnectionsAndInitCachesInternal().block();
+    }
+
+    private Mono<List<OpenConnectionResponse>> openConnectionsAndInitCachesInternal() {
+        if (this.proactiveContainerInitConfig != null) {
+            return this.asyncDocumentClient
+                    .openConnectionsAndInitCaches(this.proactiveContainerInitConfig)
```

Comment:
just brainstorming here: 

what if we just utilize the method exists in CosmosContainer/CosmosAsyncContainer, for example:
```
foreach(CosmosContainerIdentity identity: this.proactiveContainerInitConfig.getIdentities()) {
     this
.getDatabase(identity.getDatabaseName)
.getContainer(identity.getContainerName)
.openConnectionsAndInitCaches(this.proactiveContainerInitConfig.getNumberOfRegions)
}
```

few benefits from the above method:
1. We discussed that concern that what if customer configured in cosmosClientBuilder and then they still call asyncContainer.openConnectionAndInitCaches -> we would not need to worry about it anymore as the flag will be set automatically
2. Probably less code change in GatewayAddressCache etc area
3. Utilizing the statistics printed out in CosmosAsyncContainer -> I think eventually it would be nice to also log some statistics about how many connections are established successfully etc

Reply:
Discussed further offline.